### PR TITLE
feat(yaml): Include memory consumption litmus book for cstor

### PR DESCRIPTION
**Pull request by dargasudarshan**
Signed-off-by: Sudarshan <sudarshan.darga@cloudbyte.com>

**What this PR does / why we need it**:
- Include memory consumption litmus book for cstor.
- Deploys a volume consuming cstor-sc which is set with resource limits.
- Runs a python script to fetch the memory consumed by cstor istgt for the dd profiles ran on the volume.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: providers/openebs/installers/storageclass/0.7/litmusbook/storageclass_setup.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
       restartPolicy: Never
       containers:
       - name: ansibletest
-        image: openebs/ansible-runner:ci
+        image: dargas/ansible-runner:ci
```
Comment (ksatchit) [inline providers/openebs/installers/storageclass/0.7/litmusbook/storageclass_setup.yaml]: image?

Comment (dargasudarshan) [reply providers/openebs/installers/storageclass/0.7/litmusbook/storageclass_setup.yaml]: done @ksatchit , updated.

```
[0;34mNo config file found; using defaults[0m
[0;34m/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected[0m
[0;34m/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected[0m

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
[0;32mok: [127.0.0.1][0m

TASK [Record test instance/run ID] *********************************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Construct testname appended with runID] **********************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [include_tasks] ***********************************************************
[0;36mincluded: /common/utils/update_litmus_result_resource.yml for 127.0.0.1[0m

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
[0;33mchanged: [127.0.0.1] => {"changed": true, "checksum": "6374b62747b012fe620c2c8b5e13c8bc0005bb9b", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "b0c37f66dd8e2b79ac865db94ca62ffa", "mode": "0644", "owner": "root", "size": 414, "src": "/root/.ansible/tmp/ansible-tmp-1542203858.28-16702580229966/source", "state": "file", "uid": 0}[0m

TASK [Analyze the cr yaml] *****************************************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.808758", "end": "2018-11-14 13:57:39.903551", "rc": 0, "start": "2018-11-14 13:57:39.094793", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: memcheck-test \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: memcheck-test ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}[0m

TASK [Apply the litmus result CR] **********************************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.332997", "end": "2018-11-14 13:57:41.415296", "failed_when_result": false, "rc": 0, "start": "2018-11-14 13:57:40.082299", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/memcheck-test configured", "stdout_lines": ["litmusresult.litmus.io/memcheck-test configured"]}[0m

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Analyze the cr yaml] *****************************************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Apply the litmus result CR] **********************************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Verify that the AUT (Application Under Test) is running] *****************
[0;33mchanged: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n app-prometheus -l name=prometheus --no-headers -o custom-columns=:status.phase", "delta": "0:00:00.935684", "end": "2018-11-14 13:57:42.650460", "rc": 0, "start": "2018-11-14 13:57:41.714776", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}[0m

TASK [Derive PV from application PVC] ******************************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc prometheus-claim -o custom-columns=:spec.volumeName -n app-prometheus --no-headers", "delta": "0:00:00.909804", "end": "2018-11-14 13:57:43.771430", "rc": 0, "start": "2018-11-14 13:57:42.861626", "stderr": "", "stderr_lines": [], "stdout": "pvc-1a1f992f-e811-11e8-9195-42010a800019", "stdout_lines": ["pvc-1a1f992f-e811-11e8-9195-42010a800019"]}[0m

TASK [Pick a cStor target pod belonging to the PV] *****************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -l openebs.io/target=cstor-target -n openebs --no-headers | grep pvc-1a1f992f-e811-11e8-9195-42010a800019 | shuf -n1 | awk '{print $1}'", "delta": "0:00:00.904328", "end": "2018-11-14 13:57:44.862577", "rc": 0, "start": "2018-11-14 13:57:43.958249", "stderr": "", "stderr_lines": [], "stdout": "pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck", "stdout_lines": ["pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck"]}[0m

TASK [Run python script for memory usage validation it takes couple of minutes] ***
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "python test-mem.py pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck openebs", "delta": "0:03:24.451622", "end": "2018-11-14 14:01:09.506751", "failed_when_result": false, "rc": 0, "start": "2018-11-14 13:57:45.055129", "stderr": "Defaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.\nDefaulting container name to cstor-istgt.\nUse 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "stderr_lines": ["Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod.", "Defaulting container name to cstor-istgt.", "Use 'kubectl describe pod/pvc-1a1f992f-e811-11e8-9195-42010a800019-target-7dc98f5576sn2ck -n openebs' to see all of the containers in this pod."], "stdout": "282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n282.234375 MB\n[282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375]\nPass", "stdout_lines": ["282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "282.234375 MB", "[282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375, 282.234375]", "Pass"]}[0m

TASK [set_fact] ****************************************************************
[0;32mok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}[0m

TASK [include_tasks] ***********************************************************
[0;36mincluded: /common/utils/update_litmus_result_resource.yml for 127.0.0.1[0m

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Analyze the cr yaml] *****************************************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Apply the litmus result CR] **********************************************
[0;36mskipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}[0m

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
[0;33mchanged: [127.0.0.1] => {"changed": true, "checksum": "56c63d288f8b46f0acdce766d34462f3dc398293", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "62ea7a8394a5f2589240a381295f4035", "mode": "0644", "owner": "root", "size": 412, "src": "/root/.ansible/tmp/ansible-tmp-1542204069.85-210176317519975/source", "state": "file", "uid": 0}[0m

TASK [Analyze the cr yaml] *****************************************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.809782", "end": "2018-11-14 14:01:12.808366", "rc": 0, "start": "2018-11-14 14:01:11.998584", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: memcheck-test \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: memcheck-test ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}[0m

TASK [Apply the litmus result CR] **********************************************
[0;33mchanged: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.001944", "end": "2018-11-14 14:01:14.002060", "failed_when_result": false, "rc": 0, "start": "2018-11-14 14:01:13.000116", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/memcheck-test configured", "stdout_lines": ["litmusresult.litmus.io/memcheck-test configured"]}[0m

PLAY RECAP *********************************************************************
[0;33m127.0.0.1[0m                  : [0;32mok=14  [0m [0;33mchanged=10  [0m unreachable=0    failed=0   

```

File: apps/prometheus/functional/memcheck/run_litmus_test.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        command: ["/bin/bash"]
+        args: ["-c", "ansible-playbook ./prometheus/functional/memcheck/test.yml -i /etc/ansible/hosts -v; exit 0"]
+        volumeMounts:
+          - name: logs 
+            mountPath: /var/log/ansible 
+        tty: true
+      - name: logger
+        image: openebs/logger
+        command: ["/bin/bash"]
+        args: ["-c", "./logger.sh -d 10 -r maya,openebs,pvc,memcheck; exit 0"]
```
Comment (ksatchit) [inline apps/prometheus/functional/memcheck/run_litmus_test.yml]: we need to be using the container name option for `-d` duration v/s the time-limit.

Comment (dargasudarshan) [reply apps/prometheus/functional/memcheck/run_litmus_test.yml]: done.

**Comment by ksatchit** [comment]
we may need to increase load for a better validation of memory limits being honored. Will take this, next set of changes can be a separate PR.

**Pull request merged**